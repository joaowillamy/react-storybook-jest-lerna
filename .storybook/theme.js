import { create } from "@storybook/theming/create";

export default create({
  base: "light",

  colorPrimary: "#FDCB13",
  colorSecondary: "#18ACC4",

  // UI
  appBg: "white",
  appContentBg: "#F2F3F4",
  appBorderColor: "#F2F3F4",
  appBorderRadius: 4,

  // Typography
  fontBase: '"Montserrat", sans-serif',
  fontCode: "monospace",

  brandTitle: "Quero front test"
  // brandUrl: 'http://resolvvi.com.br/',
  // brandImage: ''
});
