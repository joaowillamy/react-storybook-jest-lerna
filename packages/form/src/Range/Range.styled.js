import styled from "styled-components";

export const Range = styled.div`
  width: 100%;
`;

export const RangeLabel = styled.label`
  font-size: 14px;
  font-weight: bold;
  margin: 0;
  padding: 0;
`;

export const RangeValue = styled.p`
  margin: 5px 0;
`;

export const RangeInput = styled.input`
  background: ${({ theme }) => theme.colors.bluePrimary};
  border-radius: 10px;
  height: 4px;
  margin: 15px 0;
  outline: none;
  transition: opacity 0.2s;
  width: 100%;

  ::-webkit-slider-thumb {
    appearance: none;
    background: #fff;
    border: 2px solid ${({ theme }) => theme.colors.bluePrimary};
    border-radius: 50%;
    cursor: pointer;
    height: 25px;
    width: 25px;
    padding: 10px;
  }

  ::-moz-range-thumb {
    background: #fff;
    border: 2px solid ${({ theme }) => theme.colors.bluePrimary};
    border-radius: 50%;
    cursor: pointer;
    height: 25px;
    width: 25px;
    padding: 10px;
  }
`;
