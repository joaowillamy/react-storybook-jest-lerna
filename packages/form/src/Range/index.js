import React, { useState, useCallback } from "react";
import PropTypes from "prop-types";

import * as S from "./Range.styled";

const Range = ({ label, value, minRange, maxRange, onChange }) => {
  const [currentValue, setCurrentValue] = useState(value);
  const onChangeCallback = useCallback(onChange);

  return (
    <S.Range>
      {label && <S.RangeLabel>{label}</S.RangeLabel>}
      <S.RangeValue>
        {currentValue.toLocaleString("pt-br", {
          style: "currency",
          currency: "BRL",
          minimumFractionDigits: 2,
          maximumFractionDigits: 2
        })}
      </S.RangeValue>
      <S.RangeInput
        type="range"
        min={minRange}
        max={maxRange}
        value={currentValue}
        onChange={event => {
          setCurrentValue(Number(event.target.value));
          onChangeCallback && onChangeCallback(Number(event.target.value));
        }}
      />
    </S.Range>
  );
};

Range.propTypes = {
  label: PropTypes.string,
  value: PropTypes.number,
  maxRange: PropTypes.number,
  minRange: PropTypes.number,
  onChange: PropTypes.func
};

Range.defaultProps = {
  value: 10000
};
export default Range;
