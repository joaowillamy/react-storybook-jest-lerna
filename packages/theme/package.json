{
  "name": "@will-ui-kit/theme",
  "version": "1.0.2",
  "main": "lib/index.js",
  "module": "src/index.js"
}
