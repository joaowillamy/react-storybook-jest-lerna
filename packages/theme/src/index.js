export { default as ThemeProvider } from "./ThemeProvider";
export { default as DEFAULT_THEME } from "./theme";
export { GlobalStyles } from "./GlobalStyles";
