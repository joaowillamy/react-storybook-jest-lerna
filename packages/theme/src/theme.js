const BASE = 3;

const borders = {
  sm: color => `1px solid ${color || "#00000029"}`,
  md: color => `3px solid ${color || "#00000029"}`
};

const colors = {
  bluePrimary: "#18ACC4",
  blueSecondary: "#007A8D",
  yellowPrimary: "#FDCB13",
  yellowSecondary: "#DE9E1F",
  green: "#0FA866",
  black: "#1F2D30",
  grey: "#FBFBFB",
  Overlay: "#1F2D30e0" // com 88% de opacidade.
};

const fonts = {
  Montserrat: "Montserrat"
};

const fontSizes = {
  xs: "12px",
  sm: "14px",
  md: "16px",
  lg: "18px",
  xl: "22px",
  xxl: "26px",
  xxxl: "36px"
};

const fontWeights = {
  thin: 100,
  extralight: 200,
  light: 300,
  regular: 400,
  medium: 500,
  semibold: 600,
  bold: 700,
  extrabold: 800,
  black: 900
};

const iconSizes = {
  sm: "12px",
  md: "16px",
  lg: "26px"
};

const shadows = {
  sm: color => `0 1px 3px ${color || "#00000029"}`,
  md: color => `0 3px 4px ${color || "#00000029"}`,
  lg: color => `0 3px 6px ${color || "#00000029"}`
};

const space = {
  xs: `${BASE}px`,
  sm: `${BASE * 2}px`,
  md: `${BASE * 3}px`,
  lg: `${BASE * 4}px`,
  xl: `${BASE * 8}px`,
  xxl: `${BASE * 10}px`,
  xxxl: `${BASE * 12}px`
};

const DEFAULT_THEME = {
  borders,
  colors,
  fonts,
  fontSizes,
  fontWeights,
  iconSizes,
  shadows,
  space
};

export default DEFAULT_THEME;
