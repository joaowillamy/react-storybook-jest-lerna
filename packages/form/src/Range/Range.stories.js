import React from "react";
import { ThemeProvider } from "@joaowillamy-test-quero/theme";

import Range from ".";

export default {
  component: Range,
  title: "form|Range"
};

export const range = () => {
  return (
    <ThemeProvider>
      <Range
        value={10000}
        maxRange={10000}
        label={"ATÃ‰ QUANTO PODE PAGAR?"}
        onChange={data => {}}
      />
    </ThemeProvider>
  );
};
