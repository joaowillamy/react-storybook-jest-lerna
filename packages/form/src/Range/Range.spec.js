import React from "react";
import renderer from "react-test-renderer";
import "jest-styled-components";
import { ThemeProvider } from "@joaowillamy-test-quero/theme";

import Range from ".";

describe("Range", () => {
  test("Range", () => {
    const tree = renderer
      .create(
        <ThemeProvider>
          <Range />
        </ThemeProvider>
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
