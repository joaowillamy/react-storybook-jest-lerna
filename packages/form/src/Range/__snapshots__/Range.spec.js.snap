// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Range Range 1`] = `
.c0 {
  width: 100%;
}

.c1 {
  margin: 5px 0;
}

.c2 {
  background: #18ACC4;
  border-radius: 10px;
  height: 4px;
  margin: 15px 0;
  outline: none;
  -webkit-transition: opacity 0.2s;
  transition: opacity 0.2s;
  width: 100%;
}

.c2::-webkit-slider-thumb {
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  background: #fff;
  border: 2px solid #18ACC4;
  border-radius: 50%;
  cursor: pointer;
  height: 25px;
  width: 25px;
  padding: 10px;
}

.c2::-moz-range-thumb {
  background: #fff;
  border: 2px solid #18ACC4;
  border-radius: 50%;
  cursor: pointer;
  height: 25px;
  width: 25px;
  padding: 10px;
}

<div
  className="c0"
>
  <p
    className="c1"
  >
    R$Â 10,000.00
  </p>
  <input
    className="c2"
    onChange={[Function]}
    type="range"
    value={10000}
  />
</div>
`;
